python/setup.py

```diff
@@ -90,14 +90,14 @@ def run(self):
     copy2(SCRIPT_PATH, SCRIPT_TARGET)
 
     install_requires = [
-        "numpy",
-        "typing",
-        "pandas == 1.1.4",
-        "psutil",
-        "pyarrow >= 0.10",
-        "ray == 1.4.0",
-        "pyspark >= 3.0.0, < 3.1.0",
-        "netifaces"
+        "numpy < 2.0.0",
```

Comment:
Hi @mjschock, I think the changes for `pandas` and 'ray' make sense. Why you change the others? They should be flexible already.

Reply:
Actually, ray has some internal changes for different versions. So we have to specify the given version.

Reply:
it's a preventative measure so i don't run into things like the pyspark versioning issue in the future.  assuming other library authors play nice and abide by semver then patch and minor updates shouldn't break anything, only major.  i can reduce the changes just to pyspark if desired but i wanted to be consistant

Reply:
@mjschock , based on our previous experience, we usually need to make changes to support new Ray releases like 1.5. So we should probably keep `ray == 1.4.0` otherwise things may break. For most other dependencies, my understanding is setting a minimal version or without setting anything should be fine. Can we just change pyspark and pandas for now to unblock your needs?